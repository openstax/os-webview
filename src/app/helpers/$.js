// FIX:  Delete as many of these as possible
const $ = {};

$.isTouchDevice = () => (
    ('ontouchstart' in window) ||
    (navigator.MaxTouchPoints > 0) ||
    (navigator.msMaxTouchPoints > 0)
 );

$.stringCompare = (a, b) => (a < b) ? -1 : +(a > b);

$.lowerCaseCompare = (a, b) => $.stringCompare(a.toLowerCase(), b.toLowerCase());

const tick = 1000 / 40;
const spaceForMenu = 59;
const targetStep = 200;
const targetTicks = 20;

$.scrollTo = (el, offset = 0) => {
    const rect = el.getBoundingClientRect();
    const offsetTop = rect.top - spaceForMenu - offset;
    const direction = Math.sign(offsetTop);
    let magnitude = Math.abs(offsetTop);
    const chosenStep = (targetStep + magnitude / targetTicks) / 2;

    const i = setInterval(() => {
        const step = (magnitude > chosenStep) ? chosenStep : magnitude;
        const scrollBody = document.documentElement.scrollTop || document.body.scrollTop;

        window.scrollTo(0, scrollBody + direction * step);
        magnitude -= step;
        if (magnitude <= 0) {
            clearInterval(i);
        }
    }, tick);
};

$.applyScrollFix = (view) => {
    let freezePosition = null;
    const setFreezePosition = () => {
        freezePosition = document.documentElement.scrollTop || document.body.scrollTop;
    };
    const handleMouseLeave = (e) => {
        if (!e.currentTarget.contains(e.relatedTarget)) {
            freezePosition = null;
        }
    };
    const handleWheelEvent = (e) => {
        const el = e.currentTarget;
        const delta = e.deltaY || e.wheelDelta / 4;

        el.scrollTop = el.scrollTop + delta;
        e.preventDefault();
    };
    const touchOutside = (el) => (
        (e) => {
            if (!el.contains(e.target)) {
                freezePosition = null;
            }
        }
    );
    let scrollStart;
    const setScrollStart = (e) => {
        scrollStart = e.currentTarget.scrollTop + e.touches[0].pageY;
    };
    const scroll = (e) => {
        e.currentTarget.scrollTop = scrollStart - event.touches[0].pageY;
        event.preventDefault();
    };

    for (const el of view.el.querySelectorAll('.mac-scroll')) {
        view.attachListenerTo(el, 'scroll', setFreezePosition);
        view.attachListenerTo(el, 'mouseleave', handleMouseLeave);
        view.attachListenerTo(el, 'mousewheel', handleWheelEvent);
        view.attachListenerTo(window, 'touchstart', touchOutside(el));
        if ($.isTouchDevice() && window.innerWidth < 769) {
            view.attachListenerTo(el, 'touchstart', setScrollStart);
            view.attachListenerTo(el, 'touchmove', scroll);
        }
    }
    view.attachListenerTo(window, 'scroll', () => {
        if (freezePosition !== null) {
            window.scrollTo(0, freezePosition);
        }
    });
};

$.hashClick = (event, options = {doHistory: true}) => {
    const node = event.delegateTarget;
    const destUrl = `${node.pathname}${node.hash}`;
    const targetEl = document.getElementById(node.hash.substr(1));

    $.scrollTo(targetEl);
    if (options.doHistory) {
        history.pushState({}, '', destUrl);
    }
    event.preventDefault();
};

const invalidEmailPatterns = [
    /@(aol|gmail|hotmail|yahoo).com/i
];

$.testInstitutionalEmail = (element) => {
    const ieValue = element.value;

    for (const re of invalidEmailPatterns) {
        if (re.test(ieValue)) {
            return false;
        }
    }

    return true;
};

$.htmlToText = (html) => {
    const temp = document.createElement('div');

    temp.innerHTML = html;
    return temp.textContent;
};

$.parseSearchString = (searchString) => {
    const result = {};

    searchString.substr(1).split('&').forEach((item) => {
        const [k, v] = item.split('=');

        (k in result) ? result[k].push(decodeURIComponent(v)) : result[k] = [decodeURIComponent(v)];
    });
    return result;
};

export default $;
