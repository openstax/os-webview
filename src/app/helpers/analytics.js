import settings from 'settings';
import linkHelper from '~/helpers/link';

const RELATIVE_TO_ROOT = /^\//;
const SETUP_GTM = Symbol();
const SETUP_GA = Symbol();

class Analytics {

    constructor() {
        this.start();
    }

    start() {
        this[SETUP_GA]();
        this[SETUP_GTM]();
    }

    send(fields) {
        if (this.tracking) {
            SystemJS.import('ga').then(() => {
                window.ga('send', fields);
            });
        }
    }

    sendPageview(page) {
        let frag = page || location.pathname;

        if (!(RELATIVE_TO_ROOT).test(frag)) {
            frag = `/${frag}`;
        }

        this.send({
            hitType: 'pageview',
            page: frag
        });
    }

    sendEvent(fields) {
        this.send(Object.assign(
            {hitType: 'event'},
            fields
        ));
    }

    record(href) {
        if (linkHelper.isPDF(href)) {
            this.sendEvent({
                eventCategory: 'PDF',
                eventAction: 'download',
                eventLabel: href
            });
        }

        if (linkHelper.isExternal(href)) {
            this.sendEvent({
                eventCategory: 'External',
                eventAction: 'open',
                eventLabel: href
            });
        }
    }

    [SETUP_GTM]() {
        (function (w, d, s, l, i) {
            // Disable ESLint rules since we're copying Google's script
            /* eslint max-params: 0 */
            /* eslint eqeqeq: 0 */
            /* eslint prefer-template: 0 */

            w[l] = w[l] || [];
            w[l].push({
                'gtm.start': new Date().getTime(),
                event: 'gtm.js'
            });

            const f = d.getElementsByTagName(s)[0];
            const j = d.createElement(s);
            const dl = l !== 'dataLayer' ? '&l=' + l : '';

            j.async = true;
            j.src = '//www.googletagmanager.com/gtm.js?id=' + i + dl;
            f.parentNode.insertBefore(j, f);
        })(window, document, 'script', 'dataLayer', settings.tagManagerID);
    }

    [SETUP_GA]() {
        if (typeof window.ga !== 'function') {
            window.GoogleAnalyticsObject = 'ga';
            window.ga = {
                q: [['create', settings.analyticsID, 'auto']],
                l: Date.now()
            };
        } else {
            window.ga('create', settings.analyticsID, 'auto');
        }

        document.addEventListener('submit', (e) => {
            if (typeof e.target !== 'object' || typeof e.target.action !== 'string') {
                return;
            }

            if (e.target.action.indexOf('https://www.salesforce.com/') === 0) {
                const formData = new FormData(e.target);

                this.sendEvent({
                    eventCategory: 'Salesforce',
                    eventAction: 'submit',
                    eventLabel: formData.get('lead_source')
                });
            }
        });
    }

}

const analytics = new Analytics();

export default analytics;
