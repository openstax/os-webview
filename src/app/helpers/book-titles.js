let comingSoon = [
    'Astronomy',
    'Elementary Algebra',
    'Intermediate Algebra',
    'Microbiology',
    {
        text: 'University Physics',
        value: 'University Physics (Calc)'
    }
];

export let published = [
    'Algebra and Trigonometry',
    'American Government',
    'Anatomy & Physiology',
    'Biology',
    'Calculus',
    'Chemistry',
    {
        text: 'Chemistry: Atoms First',
        value: 'Chem: Atoms First'
    },
    'College Algebra',
    {
        text: 'College Physics',
        value: 'College Physics (Algebra)'
    },
    {
        text: 'College Physics for AP® Courses',
        value: 'AP Physics'
    },
    {
        text: 'Concepts of Biology',
        value: 'Concepts of Bio (non-majors)'
    },
    'Introduction to Sociology',
    {
        text: 'Introduction to Sociology 2e',
        value: 'Introduction to Sociology'
    },
    'Introductory Statistics',
    {
        text: 'Prealgebra',
        value: 'PreAlgebra'
    },
    {
        text: 'Precalculus',
        value: 'Precalc'
    },
    {
        text: 'Principles of Economics',
        value: 'Economics'
    },
    {
        text: 'Principles of Macroeconomics',
        value: 'Macro Econ'
    },
    {
        text: 'Principles of Macroeconomics for AP® Courses',
        value: 'AP Macro Econ'
    },
    {
        text: 'Principles of Microeconomics',
        value: 'Micro Econ'
    },
    {
        text: 'Principles of Microeconomics for AP® Courses',
        value: 'AP Micro Econ'
    },
    'Psychology',
    {
        text: 'U.S. History',
        value: 'US History'
    }
];

let bookTitles = [...comingSoon, ...published].sort((a, b) => {
    let aText = a.text || a,
        bText = b.text || b;

    return aText < bText ? -1 : aText !== bText;
});

export default bookTitles;
