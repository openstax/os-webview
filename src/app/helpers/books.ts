export type Book = {
    book_state: string;
    cover_url: string;
    salesforce_abbreviation: string;
    salesforce_name: string;
    slug: string;
    subjects: string[];
    title: string;
};

const statesToInclude = ['live', 'new_edition_available', 'coming_soon'];

export const filterLiveBooks = (entry: Book) =>
    statesToInclude.includes(entry.book_state);

function BookToSalesforceBook(book: Book) {
    return {
        text: book.salesforce_name,
        value: book.salesforce_abbreviation,
        comingSoon: book.book_state === 'coming_soon',
        subjects: book.subjects,
        coverUrl: book.cover_url
    };
}

export type SalesforceBook = ReturnType<typeof BookToSalesforceBook>;

export function salesforceTitles(books: Book[]) {
    const seenTitles: {[key: string]: boolean} = {};

    return books
        .filter(filterLiveBooks)
        .filter((book) => {
            const abbrev = book.salesforce_abbreviation;
            const seen = abbrev in seenTitles;

            seenTitles[abbrev] = true;
            return abbrev && !seen;
        })
        .map(BookToSalesforceBook)
        .sort((a, b) => (a.text < b.text ? -1 : 1));
}

export function subjects(sfTitles: Book[]) {
    return sfTitles
        .reduce<string[]>((a, b) => a.concat(b.subjects || []), [])
        .reduce<string[]>((a, b) => (a.includes(b) ? a : a.concat(b)), []);
}
