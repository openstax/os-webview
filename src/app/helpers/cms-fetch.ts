import retry from '~/helpers/retry';
import urlFromSlug from './url-from-slug';

export default async function cmsFetch(path) {
    const url = path.replace(/[^?]+/, urlFromSlug);

    try {
        return (await retry(() => fetch(url))).json();
    } catch (err) {
        return Promise.reject(new Error(`Failed to fetch ${path}: ${err}`));
    }
}

export async function cmsPost(path, payload, method='POST') {
    const url = path.replace(/[^?]+/, urlFromSlug);
    const params = new window.URLSearchParams(payload);
    const qs = method.toLowerCase() === 'delete' ? `?${params}` : '';

    try {
        return (await fetch(`${url}${qs}`, {
            method,
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(payload)
        })).json();
    } catch (e) {
        return Promise.reject(new Error(`Failed to ${method} ${url}${qs}`));
    }
}
