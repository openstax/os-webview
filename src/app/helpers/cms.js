import settings from 'settings';

const jsonToQueryString = (json) =>
    Object.keys(json).map((key) =>
        `${encodeURIComponent(key)}=${encodeURIComponent(json[key])}`
    ).join('&');

const cms = {
    query: (searchOptions) => {
        const url = `${settings.apiOrigin}/api/v1/pages/?format=json&${jsonToQueryString(searchOptions)}`;

        return new Promise((resolve) => fetch(url)
        .then((response) => response.json())
        .then(resolve));
    }
};

export default cms;
