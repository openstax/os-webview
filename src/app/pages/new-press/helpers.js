import React from 'react';
import Byline from '~/components/byline/byline';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faExternalLinkAlt} from '@fortawesome/free-solid-svg-icons/faExternalLinkAlt';

export function ContentBlock({title, children}) {
    return (
        <div className='content-block'>
            <h2>{title}</h2>
            {children}
        </div>
    );
}

export function asDate(dateStr) {
    const [year, month, day] = dateStr.split('-');

    return new Date(year, month - 1, day);
}

export function convertedDate(dateStr) {
    return asDate(dateStr).toLocaleString('en-US', {
        month: 'short',
        day: 'numeric',
        year: 'numeric'
    });
}

function OptionalExcerpt({excerpt, url}) {
    return (
        excerpt && (
            <div className='excerpt'>
                {excerpt}â€¦ <a href={url}>Continue reading</a>
            </div>
        )
    );
}

export function PressExcerpt({
    iconUrl,
    author,
    date,
    source,
    url,
    headline,
    excerpt
}) {
    const classList = ['press-excerpt'];

    if (iconUrl) {
        classList.push('with-icon');
    }

    return (
        <div className={classList.join(' ')}>
            {iconUrl && <img src={iconUrl} alt='' />}
            <Byline author={author} date={date} source={source} />
            <div className='headline'>
                <a href={url}>
                    {headline}{' '}
                    {source && <FontAwesomeIcon icon={faExternalLinkAlt} />}
                </a>
            </div>
            <OptionalExcerpt excerpt={excerpt} url={url} />
        </div>
    );
}
