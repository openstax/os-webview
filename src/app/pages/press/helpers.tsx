import React from 'react';
import Byline from '~/components/byline/byline';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faExternalLinkAlt} from '@fortawesome/free-solid-svg-icons/faExternalLinkAlt';

export function ContentBlock({title, children}: React.PropsWithChildren<{title: string}>) {
    return (
        <div className='content-block'>
            <h2>{title}</h2>
            {children}
        </div>
    );
}

export function asDate(dateStr: string) {
    const [year, month, day] = dateStr.split('-');

    return new Date(+year, +month - 1, +day);
}

export function convertedDate(dateStr: string) {
    return asDate(dateStr).toLocaleString('en-US', {
        month: 'short',
        day: 'numeric',
        year: 'numeric'
    });
}

function OptionalExcerpt({excerpt, url}: {excerpt?: string; url: string}) {
    return (
        excerpt && (
            <div className='excerpt'>
                {excerpt}â€¦ <a href={url}>Continue reading</a>
            </div>
        )
    );
}

export function PressExcerpt({
    iconUrl,
    author,
    date,
    source,
    url,
    headline,
    excerpt
}: Parameters<typeof Byline>[0] & {
    iconUrl?: string;
    url: string;
    headline: string;
    excerpt?: string;
}) {
    const classList = ['press-excerpt'];

    if (iconUrl) {
        classList.push('with-icon');
    }

    return (
        <div className={classList.join(' ')}>
            {iconUrl && <img src={iconUrl} alt='' />}
            <Byline author={author} date={date} source={source} />
            <div className='headline'>
                <a href={url}>
                    {headline}{' '}
                    {source && <FontAwesomeIcon icon={faExternalLinkAlt} />}
                </a>
            </div>
            <OptionalExcerpt excerpt={excerpt} url={url} />
        </div>
    );
}
