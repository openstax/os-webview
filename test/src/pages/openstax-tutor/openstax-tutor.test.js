import '../../../helpers/fetch-mocker';
import Tutor from '~/pages/openstax-tutor/openstax-tutor';
import {clickElement} from '../../../test-utils';
import instanceReady from '../../../helpers/instance-ready';

describe('OpenStaxTutor', () => {
    const {instance:p, ready} = instanceReady(Tutor);

    it('matches snapshot', () => {
        return ready.then(() => {
            console.log('To update snapshot: node_modules/.bin/jest --updateSnapshot --testNamePattern=OpenStaxTutor');
            expect(p.el.innerHTML).toMatchSnapshot();
        });
    });
    it('has toggle-able FAQ entries', () => {
        return ready.then(() => {
            const toggle = p.el.querySelector('.toggled-item[role="button"]');
            const hiddenAnswer = () => toggle.parentNode.querySelector('.toggled-item.hidden');

            expect(toggle).toBeTruthy();
            expect(hiddenAnswer()).toBeTruthy();
            clickElement(toggle);
            expect(hiddenAnswer()).toBeNull();
        });
    });
});
