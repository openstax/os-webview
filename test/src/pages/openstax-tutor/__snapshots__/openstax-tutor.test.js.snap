// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenStaxTutor matches snapshot 1`] = `"<div class=\\"floating-tools\\"><div class=\\"section-navigator\\"><div class=\\"hotspot selected\\" data-id=\\"new-frontier\\"><span class=\\"dot\\"></span></div><div class=\\"hotspot null\\" data-id=\\"how-it-works\\"><span class=\\"dot\\"></span></div><div class=\\"hotspot null\\" data-id=\\"what-students-get\\"><span class=\\"dot\\"></span></div><div class=\\"hotspot null\\" data-id=\\"feature-matrix\\"><span class=\\"dot\\"></span></div><div class=\\"hotspot null\\" data-id=\\"where-money-goes\\"><span class=\\"dot\\"></span></div><div class=\\"hotspot null\\" data-id=\\"science\\"><span class=\\"dot\\"></span></div><div class=\\"hotspot null\\" data-id=\\"faq\\"><span class=\\"dot\\"></span></div><div class=\\"hotspot null\\" data-id=\\"learn-more\\"><span class=\\"dot\\"></span></div></div><div class=\\"pulsing-dot\\"><div class=\\"dot \\"></div><div class=\\"pulse\\">!</div><div class=\\"popup\\" data-html=\\"html\\"><p>OpenStax Tutor Beta is not available for high school or international courses at this time.</p></div></div></div><div class=\\"sticky-footer-region\\"><div class=\\"sticky-footer\\"><div class=\\"button-group\\"><a class=\\"btn medium\\" href=\\"https://tutor.openstax.org/dashboard\\">Get Started</a><div class=\\"description\\">Preview and create a course.</div></div><div class=\\"button-group\\"><div class=\\"description\\">Get advice from pioneers of the tool.</div><a class=\\"btn medium\\" href=\\"https://openstax.org/blog/upcoming-openstax-webinars\\">Join A Webinar</a></div></div></div><section id=\\"new-frontier\\"><div class=\\"background-imagery\\"><div class=\\"top-image\\"></div><div class=\\"middle-image\\"></div><div class=\\"bottom-image\\"></div></div><div class=\\"text-content\\"><div class=\\"above-head smaller\\"><div>Using OpenStax Tutor Beta?</div><a class=\\"btn small\\" href=\\"https://tutor.openstax.org/dashboard\\">Access Your Course</a></div><div class=\\"headline\\"><img id=\\"tutor-logo\\" alt=\\"OpenStax Tutor beta logo\\" src=\\"/images/openstax-tutor/tutor-logo.svg\\"><div class=\\"headline-text\\"><h1>Discover a new frontier in education.</h1><div class=\\"blurb\\">Improve how your students learn with research-based technology — for only $10.</div><div class=\\"blurb\\" data-html=\\"frontier.description\\"><p>OpenStax Tutor Beta gives college students the easy-to-use online courseware and learning tools they need to complete their course the first time around. And it’s all based on science. Available for College Physics, Biology 2e, and Introduction to Sociology 2e for college courses in the U.S.<br></p></div><a class=\\"btn primary\\" href=\\"#how-it-works\\">Learn More</a></div></div></div></section><section id=\\"how-it-works\\"><div id=\\"how-it-works-target\\"></div><div class=\\"text-content\\"><h1 class=\\"boxed\\">High tech. Low cost. Improved learning.</h1><h2>How does OpenStax Tutor Beta work?</h2><div class=\\"description\\" data-html=\\"howItWorks.description\\"><p>Wish you could work with every student, individually? With OpenStax Tutor Beta, you’re getting warmer. This low-cost, easy-to-use, professionally written, peer-reviewed, and research-based tool is designed to help students learn and retain information. OpenStax Tutor Beta uses OpenStax textbooks to deliver high-quality content, simulations and videos, spaced practice, and immediate feedback.</p></div></div><div class=\\"blurb-table boxed\\"><div class=\\"blurb\\"><div class=\\"icon\\"><img alt=\\"Spaced practice icon\\" src=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/documents/icon-space-practice-new.svg\\"></div><h3>Spaced practice</h3><div class=\\"smaller\\">Helps students remember what they previously learned</div></div><div class=\\"blurb\\"><div class=\\"icon\\"><img alt=\\"Personalized questions icon\\" src=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/documents/icon-personalized-q-new.svg\\"></div><h3>Personalized questions</h3><div class=\\"smaller\\">Help students learn where they need it most</div></div><div class=\\"blurb\\"><div class=\\"icon\\"><img alt=\\"Two-step questions icon\\" src=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/documents/icon-two-step-new.svg\\"></div><h3>Two-step questions</h3><div class=\\"smaller\\">Help students study more effectively</div></div><div class=\\"blurb\\"><div class=\\"icon\\"><img alt=\\"Low cost icon\\" src=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/documents/icon-low-cost-new.svg\\"></div><h3>Low cost</h3><div class=\\"smaller\\">$10 per course saves students money</div></div></div></section><section id=\\"what-students-get\\"><div class=\\"background-imagery\\"></div><div class=\\"text-content\\"><h1>See what your students will see</h1><div class=\\"description\\" data-html=\\"whatStudentsGet.description\\"><p>Check out these short, helpful videos to understand how students will see questions, reading assignments, and homework in OpenStax Tutor Beta. Get a peek at the student dashboard and analytics, too.</p></div></div><div class=\\"boxed\\"><div class=\\"carousel\\"><div class=\\"viewer\\"><div class=\\"viewport\\"><img alt=\\"The dashboard gives students an overview of their assignments and progress.\\" src=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/marketing_videos/1st_-_Student_Dashboard_with_HL.png\\"></div></div><div class=\\"video-description\\"><div class=\\"content\\" data-html=\\"whatStudentsGet.currentImage.description\\">The dashboard gives students an overview of their assignments and progress.</div></div><div class=\\"thumbnails\\"><div class=\\"active\\"><span data-html=\\"whatStudentsGet.images[0].title\\">Assignment overview</span></div><div class=\\"\\"><span data-html=\\"whatStudentsGet.images[1].title\\">Late assignments</span></div><div class=\\"\\"><span data-html=\\"whatStudentsGet.images[2].title\\">Late work</span></div><div class=\\"\\"><span data-html=\\"whatStudentsGet.images[3].title\\">Browse book</span></div><div class=\\"\\"><span data-html=\\"whatStudentsGet.images[4].title\\">Performance forecast</span></div><div class=\\"\\"><span data-html=\\"whatStudentsGet.images[5].title\\">Personalized questions</span></div><div class=\\"\\"><span data-html=\\"whatStudentsGet.images[6].title\\">Two-step questions</span></div><div class=\\"\\"><span data-html=\\"whatStudentsGet.images[7].title\\">Homework assignments</span></div><div class=\\"\\"><span data-html=\\"whatStudentsGet.images[8].title\\">Retrieval practice</span></div><div class=\\"\\"><span data-html=\\"whatStudentsGet.images[9].title\\">Immediate feedback</span></div></div></div></div></section><section id=\\"feature-matrix\\"><div class=\\"text-content two-cells\\"><div class=\\"cell\\"><h1 class=\\"small-screen\\">Our current features and future plans</h1><div class=\\"flexrow\\"><i class=\\"fa fa-check-circle true\\"></i><div>Low cost</div></div><div class=\\"flexrow\\"><i class=\\"fa fa-check-circle true\\"></i><div>Easy to build assignments</div></div><div class=\\"flexrow\\"><i class=\\"fa fa-check-circle true\\"></i><div>Personalized questions</div></div><div class=\\"flexrow\\"><i class=\\"fa fa-check-circle true\\"></i><div>Integrated digital textbook</div></div><div class=\\"flexrow\\"><i class=\\"fa fa-check-circle true\\"></i><div>Permanent access to textbook content</div></div><div class=\\"flexrow\\"><i class=\\"fa fa-check-circle true\\"></i><div>Video</div></div><div class=\\"flexrow\\"><i class=\\"fa fa-check-circle true\\"></i><div>Tutorials*</div></div><div class=\\"flexrow\\"><i class=\\"fa fa-check-circle true\\"></i><div>Student performance forecast</div></div><div class=\\"flexrow\\"><i class=\\"fa fa-check-circle true\\"></i><div>Spaced practice review questions</div></div><div class=\\"flexrow\\"><i class=\\"fa fa-check-circle true\\"></i><div>Ability to exclude questions</div></div><div class=\\"flexrow\\"><i class=\\"fa fa-check-circle true\\"></i><div>Student performance analytics</div></div><div class=\\"flexrow\\"><i class=\\"fa fa-check-circle false\\"></i><div>Ability to add own questions</div></div><div class=\\"flexrow caption smaller\\"><i class=\\"fa fa-check-circle spacer\\"></i> *Available for OpenStax Tutor Beta for College Physics</div></div><div class=\\"cell\\"><h1 class=\\"big-screen\\">Our current features and future plans</h1><div class=\\"description\\" data-html=\\"featureMatrix.description\\"><p>Thousands of students have piloted OpenStax Tutor Beta. Here are the features we’ve prioritized, and more are on the way. Have suggestions for future development? Send us an email at info@openstaxtutor.org.</p></div><div class=\\"box\\"><h2>Now available</h2><div class=\\"future-text\\" data-html=\\"comingSoon.description\\"><p><br>LMS integration by request<br><br>OpenStax Tutor Beta for Biology 2e<br><br>Highlighting and annotating for Biology 2e and Introduction to Sociology 2e<br></p><p><br></p></div><img class=\\"stamp\\" src=\\"/images/openstax-tutor/icon-megaphone.svg\\"><img class=\\"bottom-edge\\" src=\\"/images/openstax-tutor/cs-background.svg\\"><div class=\\"gradient-strip\\"></div></div></div></div></section><section id=\\"where-money-goes\\"><div class=\\"text-content\\"><h1>$10 goes a lot farther than it used to</h1><div class=\\"section-description\\" data-html=\\"whereMoneyGoes.description\\"><p>Generous grants from the Bill &amp; Melinda Gates Foundation, the Laura and John Arnold Foundation, and the Maxfield Foundation allowed us to create OpenStax Tutor Beta. And since OpenStax Tutor Beta technology is so easy to use, we don’t need salespeople. We keep our costs low and provide OpenStax Tutor Beta for $10 per student, per course, allowing us to support the tool and make research-backed improvements.</p></div></div><img class=\\"boxed big-screen\\" src=\\"/images/openstax-tutor/ten-dollars.svg\\" alt=\\"Diagram of how we spend money\\" aria-describedby=\\"breakdown-description\\"><div id=\\"breakdown-description\\" style=\\"visibility: hidden; height: 0px;\\">$5 pays for our engineers and researchers. $3 pays for authors to keep content current. $1 pays for our Support team. $1 helps us keep the lights on.</div><div class=\\"boxed small-screen\\"><img src=\\"/images/openstax-tutor/ten-dollars-mobile@2x.png\\"><div class=\\"box\\"><div class=\\"flexrow\\"><div class=\\"dot dot-1\\"></div><div>Pays for our engineers and researchers</div></div><div class=\\"flexrow\\"><div class=\\"dot dot-2\\"></div><div>Pays for authors to keep content current</div></div><div class=\\"flexrow\\"><div class=\\"dot dot-3\\"></div><div>Pays for our Support team</div></div><div class=\\"flexrow\\"><div class=\\"dot dot-4\\"></div><div>Helps us keep the lights on</div></div></div></div></section><section id=\\"science\\"><div id=\\"particles\\"></div><div class=\\"text-content\\"><h1>Behind the science of OpenStax Tutor Beta</h1><div data-html=\\"science.description\\"><p>Developed by researchers at Rice University, OpenStax Tutor Beta uses proven cognitive science principles and machine learning development to transform the way your students learn.</p></div></div></section><div class=\\"divider\\"><img src=\\"/images/openstax-tutor/faq-top-background.svg\\"></div><section id=\\"faq\\"><div class=\\"boxed\\"><h1>Frequently asked questions</h1></div><div class=\\"questions text-content\\"><div class=\\"smaller qna\\"><div class=\\"toggled-item\\" role=\\"button\\" data-index=\\"0\\"><div class=\\"question\\" data-html=\\"faq.items[0].question\\"><p>How much time does it take to set up my course in OpenStax Tutor Beta?</p></div><div class=\\"toggler\\"><i class=\\"fa fa-chevron-down\\"></i></div></div><div class=\\"toggled-item hidden\\"><div class=\\"answer\\" data-html=\\"faq.items[0].answer\\"><p>It takes 10-20 minutes to set up your OpenStax Tutor Beta account and create your course within the tool once you have your verified instructor account. Detailed page tips walk you through each feature of the tool step by step to help you hit the ground running. We manually verify every instructor account to make sure students don’t get access to instructor materials, so if you don’t yet have an instructor account with OpenStax, expect to wait 3-5 days for our Support Team to verify you.</p></div></div></div><div class=\\"smaller qna\\"><div class=\\"toggled-item\\" role=\\"button\\" data-index=\\"1\\"><div class=\\"question\\" data-html=\\"faq.items[1].question\\"><p>Does it work with my LMS?</p></div><div class=\\"toggler\\"><i class=\\"fa fa-chevron-down\\"></i></div></div><div class=\\"toggled-item hidden\\"><div class=\\"answer\\" data-html=\\"faq.items[1].answer\\"><p>OpenStax Tutor Beta allows you to export your scores into a spreadsheet to be uploaded into your learning management system. We are working towards full LMS integration for fall 2018.</p></div></div></div><div class=\\"smaller qna\\"><div class=\\"toggled-item\\" role=\\"button\\" data-index=\\"2\\"><div class=\\"question\\" data-html=\\"faq.items[2].question\\"><p>How much does it cost?</p></div><div class=\\"toggler\\"><i class=\\"fa fa-chevron-down\\"></i></div></div><div class=\\"toggled-item hidden\\"><div class=\\"answer\\" data-html=\\"faq.items[2].answer\\"><p>OpenStax Tutor Beta costs only $10 per course per semester for each student. It is important to us to keep costs low to make OpenStax Tutor Beta as accessible as possible while covering support and research-backed improvements for the tool.<br></p></div></div></div><div class=\\"smaller qna\\"><div class=\\"toggled-item\\" role=\\"button\\" data-index=\\"3\\"><div class=\\"question\\" data-html=\\"faq.items[3].question\\"><p>Can I write my own questions?</p></div><div class=\\"toggler\\"><i class=\\"fa fa-chevron-down\\"></i></div></div><div class=\\"toggled-item hidden\\"><div class=\\"answer\\" data-html=\\"faq.items[3].answer\\"><p>OpenStax Tutor Beta chooses a personalized combination of questions for each student based on their performance. We don't currently enable you to write your own questions or edit existing questions, but you can review all possible questions in the Question Library and exclude any you do not want students to see.<br></p></div></div></div><div class=\\"smaller qna\\"><div class=\\"toggled-item\\" role=\\"button\\" data-index=\\"4\\"><div class=\\"question\\" data-html=\\"faq.items[4].question\\"><p>How long do my students have access?</p></div><div class=\\"toggler\\"><i class=\\"fa fa-chevron-down\\"></i></div></div><div class=\\"toggled-item hidden\\"><div class=\\"answer\\" data-html=\\"faq.items[4].answer\\"><p>Your students have access to OpenStax Tutor Beta for a full semester. Once the semester ends, students will always be able to access the textbook content on openstax.org.</p></div></div></div><div class=\\"smaller qna\\"><div class=\\"toggled-item\\" role=\\"button\\" data-index=\\"5\\"><div class=\\"question\\" data-html=\\"faq.items[5].question\\"><p>How do I see what students see?</p></div><div class=\\"toggler\\"><i class=\\"fa fa-chevron-down\\"></i></div></div><div class=\\"toggled-item hidden\\"><div class=\\"answer\\" data-html=\\"faq.items[5].answer\\"><p>Scroll through our images and videos above on this page to see your students’ view from within the OpenStax Tutor Beta platform. If you create a course within OpenStax Tutor Beta, you can also access helpful videos previewing the student experience from within the course.</p></div></div></div><div class=\\"see-more smaller qna\\" data-html=\\"faq.knowledgeBaseCopy\\"><p>Get more answers at our <a href=\\"https://openstax.secure.force.com/help\\">Support page</a>.</p></div></div></section><section id=\\"learn-more\\"><div class=\\"boxed\\"><h1>Pioneer a new way of teaching and learning</h1></div><div class=\\"text-content\\"><div>Learn more about how your students can learn more.</div><div class=\\"buttons-row\\"><div class=\\"button-box\\"><a class=\\"btn\\" href=\\"https://tutor.openstax.org/dashboard\\">Get Started</a><div class=\\"smaller\\">to take a tour of OpenStax Tutor and view your preview course.</div></div><div class=\\"button-box\\"><a class=\\"btn\\" href=\\"https://openstax.org/blog/upcoming-openstax-webinars\\">Join a webinar</a><div class=\\"smaller\\">to get your questions answered by pioneering OpenStax Tutor Beta users and staff.</div></div></div></div></section>"`;
